File: library/src/main/scala/com/dmanchester/playfop/sapi/PlayFopModule.scala
package com.dmanchester.playfop.sapi

import com.dmanchester.playfop.sinternal.PlayFopImpl

import play.api.Configuration
import play.api.Environment
import play.api.inject.Module

/** A Play `[[https://playframework.com/documentation/2.6.x/api/scala/index.html#play.api.inject.Module Module]]`
  * for dependency-injecting PlayFOP into Scala applications.
  */
class PlayFopModule extends Module {

  def bindings(environment: Environment, configuration: Configuration) = Seq(
    bind[PlayFop].to[PlayFopImpl]
  )
}

File: library/src/main/java/com/dmanchester/playfop/api_j/PlayFopModule.java
package com.dmanchester.playfop.api_j;

import play.api.Configuration;
import play.api.Environment;
import play.api.inject.Binding;
import play.api.inject.Module;
import scala.collection.Seq;

import com.dmanchester.playfop.internal_j.PlayFopImpl;

public class PlayFopModule extends Module {

    @Override
    public Seq<Binding<?>> bindings(Environment environment, Configuration configuration) {
        return seq(
            bind(PlayFop.class).to(PlayFopImpl.class)
        );
    }
}
